### DISCUSSION TITLE
Does the rotate_grid_vectors function work for all CMIP6 ocean grids?

### DISCUSSION THREAD

**Discussion by afrigola** [SciTools/iris#5022] [Q&A] [answered]
I would like to use the iris function rotate_grid_vectors (https://scitools-iris.readthedocs.io/en/latest/_modules/iris/analysis/_grid_angles.html#rotate_grid_vectors) to rotate velocities in the ocean, from grid to lat/lon directions. My question is whether there are any assumptions about the grid characteristics within this function that would make its use for certain ocean grids inappropriate. We need to apply the function to all CMIP6 models and look at the differences between results.

**Comment by trexfeathers** [comment, marked as answer]
Hi @afrigola, I've looked around the codebase and this statement - part of your link - holds true:

```
        Most commonly, the angles are estimated solely from grid sampling
        points, using :func:`gridcell_angles` :  This allows operation on
        complex meshes defined by two-dimensional coordinates, such as most
        ocean grids.
```

The function was introduced as part of a project to support ocean grids. You should be fine 👍

**Reply by afrigola** [reply]
Hi @trexfeathers, thanks for the reply and for taking a look at the code.
